bpf/bpf_host.c

```diff
@@ -498,13 +477,27 @@ handle_ipv4(struct __ctx_buff *ctx, __u32 secctx, const bool from_host)
 		ret = ipv4_host_policy_ingress(ctx, &remoteID);
 	if (IS_ERR(ret))
 		return ret;
-	if (skip_redirect)
-		return CTX_ACT_OK;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 #endif /* ENABLE_HOST_FIREWALL */
 
+	if (skip_redirect)
+		return CTX_ACT_OK;
+
+	if (from_host) {
+		/* If we are attached to cilium_host at egress, this will
+		 * rewrite the destination mac address to the MAC of cilium_net
```

Comment:
Nit: You likely lost this fix in a rebase:
```suggestion
		 * rewrite the destination MAC address to the MAC of cilium_net.
```